File: software/bindings/java/ExampleWaterBoiling.java
import com.tinkerforge.BrickletTemperatureIR;
import com.tinkerforge.IPConnection;

public class ExampleWaterBoiling {
	private static final String host = new String("localhost");
	private static final int port = 4223;
	private static final String UID = new String("ABC"); // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		// Create connection to brickd
		IPConnection ipcon = new IPConnection(host, port); // Can throw IOException

		BrickletTemperatureIR tir = new BrickletTemperatureIR(UID); // Create device object
		
		// Add device to ip connection
		ipcon.addDevice(tir); // Can throw IPConnection.TimeoutException
		// Don't use device before it is added to a connection
		

		// Set emissivity to 0.98 (emissivity of water)
		tir.setEmissivity((int)(0xFFFF*0.98));

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		tir.setDebouncePeriod(10000);

		// Configure threshold for "object temperature greater than 100 °C" (unit is °C/10)
		tir.setObjectTemperatureCallbackThreshold('>', (short)(100*10), (short)0);

		// Add and implement temperature reached listener 
		// (called if object temperature is greater than 100 °C)
		tir.addListener(new BrickletTemperatureIR.ObjectTemperatureReachedListener() {
			public void objectTemperatureReached(short temperature) {
				System.out.println("The surface has a temperature of " + 
				                   temperature/10.0 + " °C.");
				System.out.println("The water is boiling!");
			}
		});
		
		System.out.println("Press ctrl+c to exit");
		ipcon.joinThread();
	}
}

File: software/examples/delphi/ExampleWaterBoiling.pas
program ExampleWaterBoiling;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  Math, SysUtils, IPConnection, BrickletTemperatureIR;

type
  TExample = class
  private
    ipcon: TIPConnection;
    tir: TBrickletTemperatureIR;
  public
    procedure ReachedCB(sender: TObject; const temperature: smallint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '365'; { Change to your UID }

var
  e: TExample;

{ Callback for object temperature greater than 100 °C
  (parameter has unit °C/10) }
procedure TExample.ReachedCB(sender: TObject; const temperature: smallint);
begin
  WriteLn(Format('The surface has a temperature of %f °C', [temperature/10.0]));
  WriteLn('The water is boiling!');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create();

  { Create device object }
  tir := TBrickletTemperatureIR.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Set emissivity to 0.98 (emissivity of water) }
  tir.SetEmissivity(Floor($FFFF*0.98));

  { Get threshold callbacks with a debounce time of 10 seconds (10000ms) }
  tir.SetDebouncePeriod(10000);

  { Register threshold reached callback to procedure ReachedCB }
  tir.OnObjectTemperatureReached := {$ifdef FPC}@{$endif}ReachedCB;

  { Configure threshold for "greater than 100 °C" (unit is °C/10) }
  tir.SetObjectTemperatureCallbackThreshold('>', 100*10, 0);

  WriteLn('Press key to exit');
  ReadLn;
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
